platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt

```diff
@@ -135,9 +138,11 @@ private fun RealEditor(
 
                 applyDefaultStyle()
 
-                // Restore the state of the view with the saved state
-                setHtml(state.internalHtml)
-                setSelection(state.selection.first, state.selection.second)
+                // Set initial HTML and selection, state will be restored automatically in the view
+                if (editableText.isEmpty()) {
+                    setHtml(state.internalHtml)
+                    setSelection(state.selection.first, state.selection.second)
+                }
```

Comment:
In EXA, the selection state is no longer restored when toggling the text formatting options:

[Screen_recording_20231115_162355.webm](https://github.com/matrix-org/matrix-rich-text-editor/assets/4940864/4a8a8528-7292-4773-be8b-dbd56caedb87)

Reply:
You're right. I'm reverting this code! It's weird, because it contradicts some of my previous tests while debugging, but maybe fixing the previous bug also changed that behaviour.

Reply:
I've added an extra test for this case to help catch this in future: https://github.com/matrix-org/matrix-rich-text-editor/pull/868

Reply:
Oh cool, I was adding something similar but your tests are more thorough so let's use them instead 👍 .

Reply:
Actually, I accidentally uploaded some changes, but those also verify line count is not 1 by default, so maybe we can change the #868 PR to be based on these?

Reply:
Sure that makes sense - I can update #868 to use two lines of text.

Reply:
Sure that makes sense - I can update https://github.com/matrix-org/matrix-rich-text-editor/pull/868 to use two lines of text.